src/components/ProfileLabel.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <span>
+    <router-link :to="'/profile/' + userData._id">
+      <img :src='avatarUrl' class='avatar'/>
+    </router-link>
+    {{displayName}}
+    <slot></slot>
+  </span>
+</template>
+
+<script>
+  import { mapState } from 'vuex'
+  export default {
+    name: 'profile-label',
+    props: {
+      userData: {
+        type: Object,
+        default: function () {
+          return {
+            _id: '',
+            username: '',
+            avatarUrl: '',
+            name: ''
+          }
+        }
+      }
+    },
+    computed: {
+      ...mapState({
+        displayName () {
+          return this.userData.name || this.userData.username
```

Comment:
In theory all users should now have both of these fields be required but never hurts to have a fallback :)

Reply:
For using username as a fallback, I was putting this logic in based on what is required in the backend... but based on your comment it seems like this fallback logic is not required, but should I just leave it? 

With regards to using /profile/username - I'd prefer that as well, but we need to add a route to the backend. This should be a fairly simple addition so I could go ahead and do that, then update this PR.

Reply:
Re name + username fallback. Yeah we should update the backend eventually after testing all clients, good point. I am not sure how the mobile clients handle this.

I think it would be good to have it be done on username routes but we can merge this in and have a follow up PR if you like.